src/Orleans/Lifecycle/LifecycleObservable.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Orleans.Runtime;
+
+namespace Orleans
+{
+    public class LifecycleObservable<TStage> : ILifecycleObservable<TStage>, ILifecycleObserver
```

Comment:
Can...  Not sure about 'should'.
I was concerned that may make testing a bit harder for end users.  They'd have to duplicate logic or write another version to test anything that interacts with a lifecycle.